src/controller/UserController.ts

```diff
@@ -95,4 +99,31 @@ export default class UserController extends Controller {
       next(error)
     }
   }
+
+  static async login (request: Request, response: Response, next: NextFunction) {
+    try {
+      const data = pick(request.body, ['email', 'password'])
+      const valid = await Validator.make({
+        data,
+        rules: {
+          'email': 'required',
+          'password': 'required'
+        }
+      })
+      if (valid.fails()) {
+        throw new ValidationException({ data: valid.getErrors() })
+      }
+      const user = await UserRepository.find({ email: data.email})
+      if(!hashCompare(data.password, user.password)) {
+        throw new AppException({message: 'Password do not match'})
+      }
+      user.token = await hash((Math.random() * 1000000000).toString())
+      const updateUser = new User(user)
+      await updateUser.save()
```

Comment:
Create a repository method for this
```suggestion
      const updateUser = await new User(user).save()
```